src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.Management.tests.ps1

```diff
@@ -0,0 +1,199 @@
+﻿Import-Module -Name $PSScriptRoot\..\Arcus.Scripting.Management -ErrorAction Stop
+Import-Module -Name Az.Accounts -ErrorAction Stop
+
+InModuleScope Arcus.Scripting.Management {
+    Describe "Arcus Azure Management unit tests" {
+        Context "Remove soft deleted Azure API Management service" {
+            It "Providing an API Management name that does not exist as a soft deleted service should fail" {
+                # Arrange
+                $Name = 'unexisting-apim-instance'
+
+                Mock Get-AzCachedAccessToken -MockWith {
+                    return @{
+                        SubscriptionId = "123456"
+                        AccessToken = "accessToken"
+                    }
+                }
+
+                Mock Invoke-RestMethod {
+                    return $null
+                } -Verifiable
+
+                # Act
+                { 
+                   Remove-AzApiManagementSoftDeletedService -Name $Name
+                } | Should -Throw -ExpectedMessage "API Management instance with name '$Name' is not listed as a soft deleted service and therefore it cannot be removed"
+
+                # Assert
+                Assert-VerifiableMock
+                Assert-MockCalled Get-AzCachedAccessToken -Scope It -Times 1
+            }
```

Comment:
If it's 'not existing' APIM resources we're testing, this can be an integration test, too right?

Reply:
Good one 👍🏻 added as an integration test